### PR TITLE
Added viewpager2 in the favorites fragment, with 2 tabs

### PR DESCRIPTION
**Pull request by krishnachaitanya0107**
Added viewpager2 in the favorites fragment, with 2 tabs .

### CODE REVIEW

File: app/src/main/java/divyansh/tech/animeclassroom/favorites/DynamicTabFragment.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package divyansh.tech.animeclassroom.favorites
+
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import dagger.hilt.android.AndroidEntryPoint
+import divyansh.tech.animeclassroom.R
+
+
+@AndroidEntryPoint
+class DynamicTabFragment : Fragment(R.layout.fragment_dynamic_tab) {
```
Comment (justdvnsh) [inline app/src/main/java/divyansh/tech/animeclassroom/favorites/DynamicTabFragment.kt]: @krishnachaitanya0107 Make a new package inside the favorites package, named screens. Put this dynamic Tab Fragment there

Comment (justdvnsh) [reply app/src/main/java/divyansh/tech/animeclassroom/favorites/DynamicTabFragment.kt]: Also, add data binding layout for this tab, and include the _binding and binding variables inside of the fragment, like in favorites fragment and add onCreateView() method call

File: app/src/main/java/divyansh/tech/animeclassroom/favorites/FavoritesFragment.kt [hunk 2]
```diff
@@ -1,9 +1,129 @@
...
+    ): View? {
+
+        _favoritesFragmentBinding = FragmentFavoritesBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        pagerAdapter = ViewPagerAdapter(fragmentManager = childFragmentManager, lifecycle = lifecycle)
```
Comment (justdvnsh) [inline app/src/main/java/divyansh/tech/animeclassroom/favorites/FavoritesFragment.kt:48]: please make a new private function for this. onViewCreated() should have functions calls. It's just good practice.

File: app/src/main/java/divyansh/tech/animeclassroom/favorites/FavoritesFragment.kt [hunk 3]
```diff
@@ -1,9 +1,129 @@
...
+
+    }
+
+
+    override fun onStop() {
+        binding.viewPager2.unregisterOnPageChangeCallback(pageChangeCallback)
+        super.onStop()
+    }
+
+    private val pageChangeCallback = object : ViewPager2.OnPageChangeCallback() {
```
Comment (justdvnsh) [inline app/src/main/java/divyansh/tech/animeclassroom/favorites/FavoritesFragment.kt]: always declare all the variables at the top

File: app/src/main/java/divyansh/tech/animeclassroom/favorites/FavoritesFragment.kt [hunk 4]
```diff
@@ -1,9 +1,129 @@
...
+
+        override fun onPageSelected(position: Int) {
+            super.onPageSelected(position)
+        }
+    }
+
+    class ViewPagerAdapter( private val list: MutableList<Fragment> = mutableListOf(),
```
Comment (justdvnsh) [inline app/src/main/java/divyansh/tech/animeclassroom/favorites/FavoritesFragment.kt]: Make a new package called utils inside of the favorites directory (like it is done in home directory) and then put this class inside of the utils directory.

File: app/src/main/res/layout/fragment_dynamic_tab.xml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
```
Comment (justdvnsh) [inline app/src/main/res/layout/fragment_dynamic_tab.xml:5]: make this into a databinding layout

**Review by justdvnsh — CHANGES_REQUESTED** [review]
Hi @krishnachaitanya0107 I have reviewed this PR. Please make the required changes. Thanks.

**Comment by krishnachaitanya0107** [comment]
@justdvnsh thank you for going through the code and suggesting improvements to the code , i have implemented these changes , please go through it and do let me know if there are any more things that can be improved . I will be happy to help .

**Comment by justdvnsh** [comment]
Looks Good to me @krishnachaitanya0107 Thank you so much for your contribution. If you'd like, please join the slack channel for further discussions (You can find the link in the readme). Thanks again.

**Review by justdvnsh — APPROVED** [review]

**Comment by krishnachaitanya0107** [comment]
Thank you for accepting the pull request , it really means a lot to me . I would definitely love to join the slack channel . It felt really great working on this project and i would love to contribute and work on more issues on this project . Thank you for this wondeful opportunity .

**Comment by justdvnsh** [comment]
@krishnachaitanya0107 I am personally so happy to have contributors like you. best of luck for your open source journey. I am happy this project could have helped.

**Comment by justdvnsh** [comment]
@krishnachaitanya0107 also, I am going to launch another app in a few days. An open source app to download wallpapers for your phone. Without any ads. I would be happy if you'd contribute there as well. Moreover, this project is in heavy development, so new issues are being added everyday. Please contribute more to the project if you'd like.

**Comment by krishnachaitanya0107** [comment]
I will definitely check out that project and i will contribute to both the projects by resolving issues which i feel comfortable with . Thank you for helping me in my open source journey and for providing valuable feedback about my code to help improve my coding skills as well .

**Pull request merged**